server/controllers/users.js

```diff
@@ -73,7 +73,22 @@ export const signin = (req, res, next) => {
  *
  * This can be used right after the first login
  */
-export const updateToken = async (req, res) => {
+export const updateToken = async (req, res, next) => {
+  const { checkingTwoFactorAuth, TOTPCode } = req.body;
+
+  // the first time we try, we need to just check if the user has 2FA
+  if (req.remoteUser.twoFactorAuthToken !== null && !checkingTwoFactorAuth) {
```

Comment:
Maybe the variable `checkingTwoFactorAuth` is not useful, you could just check if user passed the `TOTPCode`.